### PR TITLE
Update Version 1.1.9

### PR DESCRIPTION
**Pull request by roomkangali**
FEATURE + BUGFIX UPDATE: "Scan Progress Visibility & Call Graph for Java/JADX". This update adds a real-time progress bar to the scan pipeline for better UX on large APKs, and repairs the Cross-Reference Context (Call Graph) feature, which was silently non-functional in the default `hybrid` / `jadx` decompiler modes.

### CODE REVIEW

File: core/engine.py [hunk 1]
```diff
@@ -333,32 +356,24 @@ def _parse_llm_response(self, response: str) -> dict:
...
+        # find('{') skips any leading ```json fence automatically, and the counter
+        # ignores braces / nested ``` fences that live inside string values (e.g. a
+        # code snippet in the "evidence" field). Running this FIRST — before any
+        # fence stripping — avoids the classic failure where a non-greedy fence regex
+        # truncates the JSON at the first ``` found inside a string value.
+        json_candidate = self._extract_json_str(response)
+
+        # Strategy 1: Fallback — strip surrounding markdown fences, then retry.
+        # Covers responses where the object isn't cleanly balanced in the raw text.
+        cleaned_response = re.sub(r'^```[a-zA-Z]*\s*\n?', '', response.strip())
```
[the remark this answers is not in the document]
Comment (mubtakir-lazuardi) [reply core/engine.py:369]: This looks like a valid edge case. However, I'd prefer to reproduce it or add a regression test before changing the extraction logic, since the current implementation intentionally prioritizes the first balanced JSON object.

**Review by mubtakir-lazuardi — COMMENTED** [review]
Looks good overall.

Applied the approved Copilot suggestions and left one comment regarding the JSON extraction flow for future investigation.

LGTM.
